services/spar/src/Spar/Scim/User.hs

```diff
@@ -664,16 +725,18 @@ assertHandleNotUsedElsewhere uid hndl = do
 -- | Helper function that translates a given brig user into a 'Scim.StoredUser', with some
 -- effects like updating the 'ManagedBy' field in brig and storing creation and update time
 -- stamps.
-synthesizeStoredUser :: UserAccount -> ST.ValidExternalId -> Scim.ScimHandler Spar (Scim.StoredUser ST.SparTag)
-synthesizeStoredUser usr veid =
+synthesizeStoredUser :: UserAccount -> Scim.ScimHandler Spar (Scim.StoredUser ST.SparTag)
+synthesizeStoredUser usr =
   logScim
     ( logFunction "Spar.Scim.User.synthesizeStoredUser"
         . logUser (userId . accountUser $ usr)
         . maybe id logHandle (userHandle . accountUser $ usr)
         . maybe id logTeam (userTeam . accountUser $ usr)
-        . maybe id logEmail (veidEmail veid)
+        . maybe id logEmail (authIdEmail =<< userAuthId (accountUser usr))
     )
     $ do
+      authId <- maybe (throwError $ Scim.serverError "Server error: user doesn't haven an AuthId") pure (authIdFromUser (accountUser usr))
```

Comment:
Is this a 5xx?  Can it happen if we search for a user by handle, and then find the user has no `AuthId`?  Not sure...

Reply:
In general authIdFromUser can only fail if 1) the user has no identity OR 2) the user has a `PhoneIdentity`